### PR TITLE
Basic Project System

### PR DESCRIPTION
**Pull request by IsotoxalDev**
Basic Project System also did some restructuring...

### CODE REVIEW

**Comment by IsotoxalDev** [comment]
This PR adds a basic project system. It also fixes some design of elements such as panel and tool button and scroll. This PR has also re structured project to suit th needs of the project system. Also deleted som unwanted files

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,6 @@ export_presets.cfg
 # Mono-specific ignores
 .mono/
 data_*/
+
+# GoDAW specific ignores
+song.gd
```
Comment (Technohacker) [inline .gitignore:12]: Is this .gitignore needed?

File: Editor/Scenes/Dialogues/TrackEditor/TrackEditor.gd [hunk 2]
```diff
@@ -1,33 +1 @@
...
-	script_editor.grab_focus()
-
-func _unpopup():
-	var file =  File.new()
-	file.open("res://song.gd", File.WRITE)
-	file.store_string(script_editor.text)
-	file.close()
-	var song: SongScript = load("res://song.gd").new()
-	song.entry()
-	emit_signal("sequence_song", song.sequence)
```
Comment (Technohacker) [inline Editor/Scenes/Dialogues/TrackEditor/TrackEditor.gd:33]: If most of the content in this file is not needed anymore you can delete it :)

File: .gitignore (hunk 1 above)
Comment (IsotoxalDev) [reply .gitignore:12]: As long as i temp save to song.gd .......

File: Editor/Scenes/Dialogues/TrackEditor/TrackEditor.gd (hunk 2 above)
Comment (IsotoxalDev) [reply Editor/Scenes/Dialogues/TrackEditor/TrackEditor.gd:33]: deleted it.

**Review by Technohacker — APPROVED** [review]

**Pull request merged**